src/AvaloniaEdit.TextMate/TextMate.cs

```diff
@@ -63,6 +63,15 @@ public void SetGrammar(string scopeName)
                 _editor.TextArea.TextView.Redraw();
             }
 
+            public void SetGrammarFile(string path)
```

Comment:
Can you please extract a common method that receives a `IGrammar` and it's used by both `SetGrammar` and `SetGrammarFile` to avoid duplicated logic?